### DISCUSSION TITLE
useXAgent问答交互提问

### DISCUSSION THREAD

**Discussion by mashupeng1** [ant-design/x#342] [Q&A]
![image](https://github.com/user-attachments/assets/54b0a728-1739-408e-ab46-ba1aa392987e)
发送实体类我会定义，但是成功调用，怎么接收response，不太会写。
我看例子都是把问的string，直接打印出来。
如果我请求是一个ChatReq实体，返回是一个ChatResp实体，怎么去定义，和使用呢？接收到ChatResp实体还要去转换成Bubble并插入BubbleList对吧？

**Comment by mashupeng1** [comment]
我是需要手动写一个请求到后端拿数据，然后填充onSuccess么？

**Comment by YumoImer** [comment]
`useXAgent` 内部预设 `XRequest` 作为请求工具，如果不满足需求的话可以使用 `request` 参数来自己实现。参考下文档：
https://x.ant.design/components/use-x-agent-cn
![image](https://github.com/user-attachments/assets/88474fca-98e8-4db5-90f5-2029a02b3fdd)
